pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -2452,7 +2457,7 @@ def request_service(self, input):
         """,
         validator=joined_validators(strict_discrete_set, strict_range),
         values=[["AUTO", "MAN"], arange(50E-6, 100)],
-        set_process=lambda v: v if isinstance(v, str) else ("%.3E" % v)
+        set_process=lambda v: v if isinstance(v, str) else ("%.3E S" % v)
```

Comment:
f-strings are the more modern style. You do not have to change it, I just want to make you aware of these possibilities:
```suggestion
        set_process=lambda v: v if isinstance(v, str) else ("{v:.3E} S")
```

For details of the formatting see https://docs.python.org/3/library/string.html#format-specification-mini-language .
Basic idea: the colon separates the variable from the formatting options.